network/p2p/tracer/gossipSubMeshTracer.go

```diff
@@ -196,6 +205,23 @@ func (t *GossipSubMeshTracer) SendRPC(rpc *pubsub.RPC, _ peer.ID) {
 	}
 }
 
+// DuplicateMessage is invoked when a duplicate message is dropped.
+func (t *GossipSubMeshTracer) DuplicateMessage(msg *pubsub.Message) {
+	peerID := msg.ReceivedFrom
+	count, err := t.duplicateMessageTrackerCache.Inc(msg.ReceivedFrom)
+	if err != nil {
+		t.logger.Err(err).
+			Bool(logging.KeyNetworkingSecurity, true).
+			Str("peer_id", peerID.String()).
```

Comment:
Please use `p2plogging.PeerId(peerID)` for this as well as other repetitions that a peer id is logged. This logger utilizes an underlying cache that addresses a known issue with encoding peer ids.